hedera-mirror-rest/__tests__/specs/balances/invalid-param-values.json

```
@@ -1,33 +1,88 @@
 {
   "description": "Balance api invalid account id format",
-  "setup": {},
-  "urls": [
-    "/api/v1/balances?account.id=lte:0.1.2.3&account.balance=gt:-2&account.publickey=-2&timestamp=-1.2&order=any&token.id=0.0.45678",
-    "/api/v1/balances?account.id=lte:0.1.2.3&account.balance=gt:9223372036854775808&account.publickey=-2&timestamp=-1.2&order=any,&token.id=0.0.45678"
-  ],
-  "responseStatus": 400,
-  "responseJson": {
-    "_status": {
-      "messages": [
-        {
-          "message": "Invalid parameter: account.id"
-        },
-        {
-          "message": "Invalid parameter: account.balance"
-        },
-        {
-          "message": "Invalid parameter: account.publickey"
-        },
-        {
-          "message": "Invalid parameter: timestamp"
-        },
-        {
-          "message": "Invalid parameter: order"
-        },
-        {
-          "message": "Unknown query parameter: token.id"
+  "setup": {
+    "accounts": [
+      {
+        "alias": "KGNABD5L3ZGSRVUCSPDR7TONZSRY3D5OMEBKQMVTD2AC6JL72HMQ",
+        "evm_address": "ac384c53f03855fa1b3616052f8ba32c6c2a2fec",
+        "num": 7
+      },
+      {
+        "num": 8
+      },
+      {
+        "evm_address": "ac384c53f03855fa1b3616052f8ba32c6c2a2fff",
+        "num": 9
+      }
+    ]
+  },
+  "tests": [
+    {
+      "urls": [
+        "/api/v1/balances?account.id=lte:0.1.2.3&account.balance=gt:-2&account.publickey=-2&timestamp=-1.2&order=any&token.id=0.0.45678",
+        "/api/v1/balances?account.id=lte:0.1.2.3&account.balance=gt:9223372036854775808&account.publickey=-2&timestamp=-1.2&order=any,&token.id=0.0.45678"
+      ],
+      "responseStatus": 400,
+      "responseJson": {
+        "_status": {
+          "messages": [
+            {
+              "message": "Invalid parameter: account.balance"
+            },
+            {
+              "message": "Invalid parameter: account.publickey"
+            },
+            {
+              "message": "Invalid parameter: timestamp"
+            },
+            {
+              "message": "Invalid parameter: order"
+            },
+            {
+              "message": "Unknown query parameter: token.id"
+            },
+            {
+              "message": "Invalid parameter: idOrAliasOrEvmAddress"
+            }
+          ]
         }
-      ]
+      }
+    },
+    {
+      "urls": [
+        "/api/v1/balances?account.id=gte:ac384c53f03855fa1b3616052f8ba32c6c2a2fec",
+        "/api/v1/balances?account.id=lt:ac384c53f03855fa1b3616052f8ba32c6c2a2fff",
+        "/api/v1/balances?account.id=lte:0.0.ac384c53f03855fa1b3616052f8ba32c6c2a2fff",
+        "/api/v1/balances?account.id=lte:KGNABD5L3ZGSRVUCSPDR7TONZSRY3D5OMEBKQMVTD2AC6JL72HMQ"
+      ],
+      "responseStatus": 400,
+      "responseJson": {
+        "_status": {
+          "messages": [
+            {
+              "message": "Invalid parameter: idOrAliasOrEvmAddress"
```

The parameter name must be the query parameter name the customer supplied so they can properly react to the bad input. It would be even better if it told you why it was invalid like unsupported lte operator for EVM address.

```suggestion
              "message": "Invalid parameter: account.id"
```

---

Thanks I have fixed this.